### DISCUSSION TITLE
How to use `run_after`?

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [stevearc/overseer.nvim#124] [Q&A] [answered]
Hi friends

I am a big fan of Overseer. Thanks a lot for making this amazing plugin. Something I have trouble getting to work is the `run_after` key when I edit a task. No matter what I put in to the `task_names` key, I get an error back about the task name not existing.

Here's an example:

<img width="1063" alt="Screenshot 2023-04-20 at 10 49 08" src="https://user-images.githubusercontent.com/22474608/233312998-c4206140-65c9-4a8f-9117-c826117b7dad.png">

I also tried putting it in double brackets to make sure it is a list of task_names, that didn't help either. I think I may have just misunderstood the docs here but I can't really figure out how to do it. Any tips?

**Comment by stevearc** [comment, marked as answer]
Hi! You have tripped over this exact TODO line
https://github.com/stevearc/overseer.nvim/blob/66c3e4433dbc9c68a969b84e694b6172fd637ad2/lua/overseer/component/run_after.lua#L13

The component editor doesn't yet support complex data types like maps. The most it can do is a list of strings, or list of ints. So the issue is that it's going to be looking for a task literally named `'{"shell", cmd = "sleep 4"}'`, which it won't find.

To get this working you can either add the component programmatically in lua, or you can define a simple named task that does what you want and put _just the name_ after the `task_names:` field.
